modules/azure/actiongroup.go

```diff
@@ -0,0 +1,57 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/resources/mgmt/insights"
+	"github.com/stretchr/testify/require"
+)
+
+// GetActionGroupResource gets the AppServicePlan.
+// ruleName - required to find the AppServicePlan.
+// resGroupName - use an empty string if you have the AZURE_RES_GROUP_NAME environment variable set
+// subscriptionId - use an empty string if you have the ARM_SUBSCRIPTION_ID environment variable set
+func GetActionGroupResource(t *testing.T, ruleName string, resGroupName string, subscriptionID string) *insights.ActionGroupResource {
+	actionGroupResource, err := getActionGroupResourceE(ruleName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return actionGroupResource
+}
+
+func getActionGroupResourceE(ruleName string, resGroupName string, subscriptionID string) (*insights.ActionGroupResource, error) {
```

Comment:
Made it public, but I am not sure I agree on making it public for the sake of tests. If the tests are in the same package as the function, then it should be able to access the private method. Do you see consumers of this using this method?

Reply:
Sorry, when I say "you might want to test" I mean a module developer may want to use this function to test for non-existence. As a general rule in terratest we always provide both the `xxxE` function that returns an error and the `xxx` function that fails the test when there is an error, and if we omit one or the other we always omit the `xxx` function. This is because you can easily derive the `xxx` function from `xxxE` (by using `require.NoError`), but it is impossible to derive `xxxE` from `xxx`.

In general, the `E` functions are necessary because of eventual consistency. There are many cases (at least in AWS) where `terraform` returns while the resource is not yet available because the underlying API is eventually consistent, and thus you need to have a retry loop to look up the resource. The `E` functions are very useful for creating those.

Reply:
Ok, that makes sense. I haven't ran into the issue of Terraform returning before the resources are created yet in Azure. But this is good to know.